### PR TITLE
Add subscription API test coverage

### PR DESCRIPTION
**Pull request by hidmic**
Depends on https://github.com/ros2/rmw/pull/256. Built on top of #115.

### CODE REVIEW

**Comment by hidmic** [comment]
CI up to `test_rmw_implementation` and `rcl`, against all RMW implementations:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11736/)
* Linux-aarch64 [Build Status](https://ci.ros2.org/job/ci_linux-aarch64/6817/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9574/)
* Windows [Build Status](https://ci.ros2.org/job/ci_windows/11676/)

File: test_rmw_implementation/test/test_publisher.cpp [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    init_options = rmw_get_zero_initialized_init_options();
+    rmw_ret_t ret = rmw_init_options_init(&init_options, rcutils_get_default_allocator());
+    ASSERT_EQ(RMW_RET_OK, ret) << rcutils_get_error_string().str;
+    init_options.enclave = rcutils_strdup("/", rcutils_get_default_allocator());
+    ASSERT_STREQ("/", init_options.enclave);
+    context = rmw_get_zero_initialized_context();
+    ret = rmw_init(&init_options, &context);
+    ASSERT_EQ(RMW_RET_OK, ret) << rcutils_get_error_string().str;
+    constexpr char node_name[] = "my_test_node";
+    constexpr char node_namespace[] = "/my_test_ns";
```
[the remark this answers is not in the document]
Comment (Blast545) [reply test_rmw_implementation/test/test_publisher.cpp:46]: > using constexpr precludes string literals entirely and reduces somewhat the size of the binary

Saw this comment in other PR, feel free to disregard my comment.

[the remark this answers is not in the document]
Comment (brawner) [reply test_rmw_implementation/test/test_publisher.cpp:46]: It's not clear to me if the temp variables are providing much utility, but it's not a big deal if you want to leave them in.

**Comment by hidmic** [comment]
Rebased on top of `master`. PTAL!

File: test_rmw_implementation/test/test_publisher.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (hidmic) [reply test_rmw_implementation/test/test_publisher.cpp]: Subjectively, I find named constants yield tests that are easier to read.

[the remark this answers is not in the document]
Comment (brawner) [reply test_rmw_implementation/test/test_publisher.cpp]: True true, which is why I'm happy to leave them in. In this case though, they are string literals which are used only once. If there is semantic content you want to convey, you can always just put it in the string if it's not going to take up too much horizontal space.

**Comment by hidmic** [comment]
@ros-pull-request-builder retest this please.

**Comment by hidmic** [comment]
Alright, all checks are passing! Merging.

**Pull request merged**